cartography/rules/spec/model.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+
+class Provider(str, Enum):
+    """Cloud and service providers that can be monitored"""
+
+    AWS = "AWS"
+    """Amazon Web Services"""
+
+    AZURE = "Azure"
+    """Microsoft Azure"""
+
+    GCP = "GCP"
+    """Google Cloud Platform"""
+
+    GITHUB = "GitHub"
+    """GitHub source code management"""
+
+    OKTA = "Okta"
+    """Okta identity and access management"""
+
+    CROSS_CLOUD = "CROSS_CLOUD"
+    """Multi-cloud or provider-agnostic rules"""
+
+
+@dataclass(frozen=True)
+class Fact:
```

Comment:
I like the Requirement `attributes` tag - I think Fact should also have an extensible field. Perhaps we could add `attributes` or `tags` to Fact as well?

Reply:
I'm thinking we leave off the attribute from the fact for now because I want to avoid duplicate data between Requirements and Facts. Right now I think it makes more sense for Requirements than Facts to have attributes. We can always add them to Facts later on.